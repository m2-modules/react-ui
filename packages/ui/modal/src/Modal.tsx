import { CSSProperties, useEffect, useRef } from "react";

export interface ModalProps {
  active?: boolean;
  background?: CSSProperties["background"];
  escapeByClick?: boolean;
  onEscape?: () => void;
  children?: React.ReactNode;
}

export default function Modal({
  active = false,
  background = "rgba(0,0,0,0.3)",
  escapeByClick = true,
  onEscape,
  children,
}: ModalProps) {
  const modalRef = useRef<HTMLElement>(null);

  useEffect(() => {
    if (!modalRef.current) return;
    if (active) {
      modalRef.current.classList.replace("opacity-0", "opacity-100");
    }
  }, [active]);

  return (
    <section
      ref={modalRef}
      className="fixed left-0 right-0 top-0 bottom-0 transform transition-opacity duration-500 opacity-0"
      style={{ background }}
      onClick={() => {
        if (!modalRef.current) return;
        if (escapeByClick) {
          modalRef.current.classList.replace("opacity-100", "opacity-0");
        }
      }}
      onTransitionEnd={() => {
        if (!modalRef.current) return;

        if (modalRef.current.classList.contains("opacity-0")) {
          onEscape?.();
        }
      }}
    >
      {children}
    </section>
  );
}
