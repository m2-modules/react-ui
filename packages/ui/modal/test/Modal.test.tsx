import React from "react";
import renderer from "react-test-renderer";
import Modal from "../src/Modal";

it("show Modal on document", () => {
  const component = renderer.create(<Modal />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
