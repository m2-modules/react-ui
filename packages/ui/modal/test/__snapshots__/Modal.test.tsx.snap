// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show Modal on document 1`] = `
<span>
  Your Input component is ready to Go!
</span>
`;
